### PR TITLE
[Reviewer: AJH] Allow an external Cassandra to be configured

### PR DESCRIPTION
**Pull request by richardwhiuk**
This change is required for the split storage work to make the homestead-prov keyspace to be provisioned correctly.

This is required for homestead-prov interface to work in a split storage environment.

### CODE REVIEW

File: homestead.root/usr/share/clearwater/cassandra-schemas/homestead_provisioning.sh [hunk 1]
```diff
@@ -1,10 +1,17 @@
...
+. /etc/clearwater/config
+
 . /usr/share/clearwater/cassandra_schema_utils.sh
 
 quit_if_no_cassandra
 
-if [[ ! -e /var/lib/cassandra/data/homestead_provisioning ]];
+CQLSH="/usr/share/clearwater/bin/run-in-signaling-namespace cqlsh $cassandra_hostname"
+
+if [[ ! -e /var/lib/cassandra/data/homestead_provisiong ]] || \
```
Comment (AlexHockey) [inline homestead.root/usr/share/clearwater/cassandra-schemas/homestead_provisioning.sh]: Typo in the database name.

File: homestead.root/usr/share/clearwater/cassandra-schemas/homestead_provisioning.sh [hunk 2]
```diff
@@ -1,10 +1,17 @@
...
+   [[ homestead_cassandra -ne "127.0.0.1" ]];
```
Comment (AlexHockey) [inline homestead.root/usr/share/clearwater/cassandra-schemas/homestead_provisioning.sh]: Missing `$`

Comment (AlexHockey) [inline homestead.root/usr/share/clearwater/cassandra-schemas/homestead_provisioning.sh]: Pretty sure the variable name is wrong as well.

Comment (AlexHockey) [inline homestead.root/usr/share/clearwater/cassandra-schemas/homestead_provisioning.sh]: And `-ne` is for integer comparison

Comment (graemerobertson) [inline homestead.root/usr/share/clearwater/cassandra-schemas/homestead_provisioning.sh]: Decent line of code all in all! The square brackets are nicely organised.

Comment (richardwhiuk) [inline homestead.root/usr/share/clearwater/cassandra-schemas/homestead_provisioning.sh]: :dromedary_camel:

**Comment by AlexHockey** [comment]
Same overall comment as in https://github.com/Metaswitch/homestead/pull/361.

Also, don't we need to tell the homestead-prov process where cassandra is?

**Comment by richardwhiuk** [comment]
Homestead-prov already uses cassandra_hostname to determine where to find cassandra - see homestead.root/usr/share/clearwater/infrastructure/scripts/homestead-prov

**Comment by AlexHockey** [comment]
Cool.

**Pull request merged**